core/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilder.java

```diff
@@ -95,6 +96,23 @@ public String getName() {
     @Override
     public abstract AggregationBuilder subAggregations(AggregatorFactories.Builder subFactories);
 
+    public final AggregationBuilder rewrite(QueryRewriteContext context) throws IOException {
+        AggregationBuilder rewritten = doRewrite(context);
+        if (rewritten == this) {
```

Comment:
hmm what if the rewritten agg needs to be rewritten? I think we need to do this in a loop as we did with the queries?